tests/unittest/test_util.py

```diff
@@ -107,6 +107,16 @@ def test_sanitize_dump_name(data, expected):
     assert sanitize_dump_name(data) == expected
 
 
+def make_fake_sleep():
+    sleeps = []
+
+    def _fake_sleep(attempt):
+        sleeps.append(attempt)
+
+    _fake_sleep.sleeps = sleeps
+    return _fake_sleep
```

Comment:
This closes over `sleeps` rather than having it as a variable in the outer environment. This is better and less error prone.